### PR TITLE
fix(zip): cd header zip64 data reading fix

### PR DESCRIPTION
**Pull request by dariaterekhova-actionengine**
I found out that it's possible that not only zip64 data can be put in the headers' extra field, but other third-party data too, so I added a mechanism that finds the start of the zip64 data among the other data in the cd header extra field

### CODE REVIEW

File: modules/zip/src/hash-file-utility.ts [hunk 1]
```diff
@@ -43,6 +43,11 @@ export async function makeHashTableFromZipHeaders(
   const hashTable: Record<string, bigint> = {};
 
   for await (const cdHeader of zipCDIterator) {
+    // Some archivers add records with size=0 for directories,
+    // so we remove it in order to shorten the hash info and fastern the search
+    if (cdHeader.compressedSize === 0n) {
```
Comment (ibgreen) [inline modules/zip/src/hash-file-utility.ts:48]: Nit: Is there a link to some additional information that compressedSize = 0 is a special case?  This looks somewhat ad-hoc and may not be completely clear / convincing to a maintainer.

File: modules/zip/src/parse-zip/cd-file-header.ts [hunk 2]
```diff
@@ -61,30 +63,62 @@ export const parseZipCDFileHeader = async (
...
+  const zip64dataList: {length: number; name: string}[] = [];
   if (uncompressedSize === BigInt(0xffffffff)) {
-    uncompressedSize = await file.getBigUint64(extraOffset + offsetInZip64Data);
-    offsetInZip64Data += 8n;
+    zip64dataList.push({name: 'uncompressedSize', length: 8});
   }
   if (compressedSize === BigInt(0xffffffff)) {
-    compressedSize = await file.getBigUint64(extraOffset + offsetInZip64Data);
-    offsetInZip64Data += 8n;
+    zip64dataList.push({name: 'compressedSize', length: 8});
```
Comment (ibgreen) [inline modules/zip/src/parse-zip/cd-file-header.ts:84]: This code is very long and is getting hard to follow. Can we break into well-named self documenting sub-functions?

**Review by ibgreen — APPROVED** [review]
Do we have a small archive that contains an example of the zero size headers so that we can add a test case?

**Review by belom88 — APPROVED** [review]

File: modules/zip/src/hash-file-utility.ts (hunk 1 above)
Comment (dariaterekhova-actionengine) [reply modules/zip/src/hash-file-utility.ts]: I gave it a thought and decided to remove it completely because compressedSize === 0 can be true not only for directory records and I can't figure out fast what's the correct way to detect them... So I'd better let it be for now
